src/gui/addnewtorrentdialog.cpp

```diff
@@ -175,6 +181,34 @@ void AddNewTorrentDialog::setTopLevel(bool value)
     SettingsStorage::instance()->storeValue(KEY_TOPLEVEL, value);
 }
 
+int AddNewTorrentDialog::savePathHistoryLength()
+{
+    return savePathHistoryLengthSetting();
+}
+
+void AddNewTorrentDialog::setSavePathHistoryLength(int value)
+{
+    Q_ASSERT(value >= minPathHistoryLength);
+    Q_ASSERT(value <= maxPathHistoryLength);
+    const int oldValue = savePathHistoryLength();
+    if (oldValue != value) {
+        savePathHistoryLengthSetting() = value;
+        settings()->storeValue(KEY_SAVEPATHHISTORY,
+                               QStringList(settings()->loadValue(KEY_SAVEPATHHISTORY, QStringList()).toStringList().mid(0, value)));
```

Comment:
Maybe `settings()->loadValue(KEY_SAVEPATHHISTORY).toStringList().mid(0, value)`?

Reply:
Do you suggest to remove the default value?

Reply:
Yes, because it is no different from default "defaultValue" of Settings::loadValue().
Also I suggest to remove redundant function-style cast (from QStringList to the same type).

Reply:
It is not a cast, but constructor. Without it:
```
../../src/gui/addnewtorrentdialog.cpp:197:82: error: no matching function for call to ‘SettingsStorage::storeValue(const QString&, QList<QString>)’
```

Reply:
And, of course, mid() is inherited method, and it returns QList, but QVariant hasn't constructor from generic QList. Then Ok.